src/actions/actions.ts

```diff
@@ -583,6 +589,21 @@ export class DeleteOperatorXVisual extends BaseOperator {
 }
 
 @RegisterAction
+export class UpperCaseOperatorXVisual extends BaseOperator {
+    public keys = ["U"];
+    public modes = [ModeName.Visual, ModeName.VisualLine];
+    
+    public async run(vimState: VimState, start: Position, end: Position): Promise<VimState> {
+      [start, end] = adjustVimSelectionPosition(vimState, start, end);
```

Comment:
When users select ranges in reverse order, we need to swap `start` and `end` first, this is contained in the extracted method so I just reuse it. 

Moreover, this method can be used for any Visual/VisualLine selection and I think it would be perfect to deal every special case if possible, then people don't need to worry about when to `getRight` or `getLeft`, whether to take current character under cursor into account or not, etc.